src/network/postgres_protocol_handler.cpp

```diff
@@ -326,6 +328,33 @@ ProcessResult PostgresProtocolHandler::ExecQueryMessage(
       ExecQueryMessageGetResult(status);
       return ProcessResult::COMPLETE;
     };
+    case QueryType::QUERY_EXPLAIN: {
+      parser::ExplainStatement *explain_stmt =
+          static_cast<parser::ExplainStatement *>(sql_stmt.get());
+      std::unique_ptr<parser::SQLStatementList> unnamed_sql_stmt_list(new parser::SQLStatementList());
+      unnamed_sql_stmt_list->PassInStatement(
+          std::move(explain_stmt->real_sql_stmt));
+      auto stmt = traffic_cop_->PrepareStatement(
+          "explain", query, std::move(unnamed_sql_stmt_list), error_message);
+      ResultType status = ResultType::UNKNOWN;
+      if (stmt) {
+        traffic_cop_->SetStatement(stmt);
+        std::vector<std::string> plan_info = StringUtil::Split(
+            planner::PlanUtil::GetInfo(stmt->GetPlanTree().get()), '\n');
+        std::vector<FieldInfo> tuple_descriptor = {
+            traffic_cop_->GetColumnFieldForValueType("Query plan",
+                                                     type::TypeId::VARCHAR)
+                                                     };
+        stmt->SetTupleDescriptor(tuple_descriptor);
+        traffic_cop_->SetResult(plan_info);
+        status = ResultType::SUCCESS;
+      } else {
+        traffic_cop_->SetErrorMessage(error_message);
+        status = ResultType::FAILURE;
+      }
+      ExecQueryMessageGetResult(status);
+      return ProcessResult::COMPLETE;
```

Comment:
Please extract this logic into a separate member function. In general, avoid large/complex switch cases.

Reply:
Can you elaborate here a bit more? This switch case for `EXPLAIN` is a bit different than other switch cases around, except only 2-3 lines matching with other switch cases, so possibility to factor out anything meaningful doesn't look obvious to me.

Reply:
The function is fairly complex as it is, the brunt being a giant complex switch statement. Ideally, each case should be a separate function, but we'll update it incrementally. For now, extract your case into a separate function.